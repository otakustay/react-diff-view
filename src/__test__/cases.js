/* In case you are looking for type of hunk, see src/Hunk/index.js */
import {parseDiff} from '..';

export const basic = `diff --git a/src/__test__/index.test.jsx b/src/__test__/index.test.jsx
index 643c2f0..7883597 100644
--- a/src/__test__/index.test.jsx
+++ b/src/__test__/index.test.jsx
@@ -21,3 +21,3 @@ describe('basic test', () => {
     test('App renders correctly', () => {
-        expect(renderer.create(<App diffText={'deff'} />).toJSON()).toMatchSnapshot();
+        expect(renderer.create(<App diffText={'diff'} />).toJSON()).toMatchSnapshot();
     });`;

export const basicHunk = parseDiff(basic)[0].hunks[0];

export const multiple = `diff --git a/src/index1.js b/src/index1.js
index e69de29..d00491f 100644
--- a/src/index1.js
+++ b/src/index1.js
@@ -0,0 +1 @@
+1
diff --git a/src/index.js b/src/index2.js
index d00491f..0cfbf08 100644
--- a/src/index2.js
+++ b/src/index2.js
@@ -1 +1 @@
-1
+2
diff --git a/src/index3.js b/src/index3.js
index 0cfbf08..e69de29 100644
--- a/src/index3.js
+++ b/src/index3.js
@@ -1 +0,0 @@
-2`;

export const multipleHunk = parseDiff(basic)[0].hunks[0];
