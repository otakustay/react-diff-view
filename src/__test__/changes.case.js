
export const normalChange = {isNormal: true, lineNumber: 0, oldLineNumber: 0};
export const insertChange = {isInsert: true, lineNumber: 0, oldLineNumber: 0};
export const deleteChange = {isDelete: true, lineNumber: 0, oldLineNumber: 0};
