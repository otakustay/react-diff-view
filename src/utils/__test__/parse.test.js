import {parseDiff} from '..';
import {basic} from '../../__test__/cases';

describe('parseDiff', () => {
    test('basic', () => {
        expect(parseDiff(basic)).toMatchSnapshot();
    });

    test('trim', () => {
        expect(parseDiff(`\n${basic}`)).toMatchSnapshot();
    });
});
