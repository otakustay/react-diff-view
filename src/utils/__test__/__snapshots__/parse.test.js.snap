// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parseDiff basic 1`] = `
Array [
  Object {
    "hunks": Array [
      Object {
        "changes": Array [
          Object {
            "content": "    test('App renders correctly', () => {",
            "isNormal": true,
            "newLineNumber": 21,
            "oldLineNumber": 21,
            "type": "normal",
          },
          Object {
            "content": "        expect(renderer.create(<App diffText={'deff'} />).toJSON()).toMatchSnapshot();",
            "isDelete": true,
            "lineNumber": 22,
            "type": "delete",
          },
          Object {
            "content": "        expect(renderer.create(<App diffText={'diff'} />).toJSON()).toMatchSnapshot();",
            "isInsert": true,
            "lineNumber": 22,
            "type": "insert",
          },
          Object {
            "content": "    });",
            "isNormal": true,
            "newLineNumber": 23,
            "oldLineNumber": 23,
            "type": "normal",
          },
        ],
        "content": "@@ -21,3 +21,3 @@ describe('basic test', () => {",
        "isPlain": false,
        "newLines": 3,
        "newStart": 21,
        "oldLines": 3,
        "oldStart": 21,
      },
    ],
    "newEndingNewLine": true,
    "newMode": "100644",
    "newPath": "src/__test__/index.test.jsx",
    "newRevision": "7883597",
    "oldEndingNewLine": true,
    "oldMode": "100644",
    "oldPath": "src/__test__/index.test.jsx",
    "oldRevision": "643c2f0",
    "type": "modify",
  },
]
`;

exports[`parseDiff trim 1`] = `
Array [
  Object {
    "hunks": Array [],
    "newEndingNewLine": true,
    "newMode": "100644",
    "newPath": "",
    "newRevision": "2222222",
    "oldEndingNewLine": true,
    "oldMode": "100644",
    "oldPath": "",
    "oldRevision": "1111111",
    "type": "modify",
  },
  Object {
    "hunks": Array [
      Object {
        "changes": Array [
          Object {
            "content": "    test('App renders correctly', () => {",
            "isNormal": true,
            "newLineNumber": 21,
            "oldLineNumber": 21,
            "type": "normal",
          },
          Object {
            "content": "        expect(renderer.create(<App diffText={'deff'} />).toJSON()).toMatchSnapshot();",
            "isDelete": true,
            "lineNumber": 22,
            "type": "delete",
          },
          Object {
            "content": "        expect(renderer.create(<App diffText={'diff'} />).toJSON()).toMatchSnapshot();",
            "isInsert": true,
            "lineNumber": 22,
            "type": "insert",
          },
          Object {
            "content": "    });",
            "isNormal": true,
            "newLineNumber": 23,
            "oldLineNumber": 23,
            "type": "normal",
          },
        ],
        "content": "@@ -21,3 +21,3 @@ describe('basic test', () => {",
        "isPlain": false,
        "newLines": 3,
        "newStart": 21,
        "oldLines": 3,
        "oldStart": 21,
      },
    ],
    "newEndingNewLine": true,
    "newMode": "100644",
    "newPath": "src/__test__/index.test.jsx",
    "newRevision": "7883597",
    "oldEndingNewLine": true,
    "oldMode": "100644",
    "oldPath": "src/__test__/index.test.jsx",
    "oldRevision": "643c2f0",
    "type": "modify",
  },
]
`;
